### ISSUE TITLE
README focuses on “what” but doesn’t answer “why”

### ISSUE THREAD

**Issue by gaffneyc** [jhawthorn/actionview_precompiler#4]
It’s not clear for someone unfamiliar with the project already (like me) why they would decide to use actionview_precompiler and what problem it is solving. For example, is it designed to improve performance? Or cold first page load time? What are the trade offs (sounds like extra memory usage)?

**Comment by krtschmr** [comment]
+1

i have no clue why to use it. also it said "on app boot", so why do i need to call the precompiler (ActionviewPrecompiler.precompile) ?

**Comment by vfonic** [comment]
+1

I'm very interested to see what's the use for this. It seems to be used for caching or? That would be great! 

Also I'm curious "how" to use it. I mean how to pass all those instances of local vars? It just mentions the call to `precompile`.

**Comment by brandoncordell** [comment]
I love the idea of this gem... at least the idea I have in my head _assuming_ I know what this does. If this is indeed about caching and speeding up rendering I’d really like to see some benchmarks vs standard rendering in Rails 6 (preferably with each of the big markup languages; erb, haml, slim, etc).

**Comment by westonganger** [comment]
+1 would love more info on the problems this gem solves

**Comment by StanBright** [comment]
Mates, I think I found the answer of most of your questions on @jhawthorn's blog https://www.johnhawthorn.com/2019/09/precompiling-rails-templates/

p.s. I agree that some parts of the blog post should be included in the README... or at least a link to the blog post :) 

p.p.s. @jhawthorn, thanks for taking the time to work on this 🙇. I'd be happy to give it a try on   [SaaSHub](https://www.saashub.com) once I've upgraded to Rails 6 (soonish).
